lib/seafoam/commands.rb

```diff
@@ -373,6 +376,56 @@ def source(name, *args)
       end
     end
 
+    # seafoam file.bgv:n describe
+    def describe(name, *args)
+      file, graph_index, *rest = parse_name(name)
+
+      if graph_index.nil? || !rest.all?(&:nil?)
+        raise ArgumentError, 'describe only works with a graph'
+      end
+      raise ArgumentError, 'describe does not take arguments' unless args.empty?
+
+      parser = BGV::BGVParser.new(file)
+      parser.read_file_header
+      parser.skip_document_props
+
+      loop do
+        index, = parser.read_graph_preheader
+        break unless index
+
+        parser.skip_graph_header
+
+        if index != graph_index
+          parser.skip_graph
+          next
+        end
+
+        graph = parser.read_graph
+        notes = Set.new
+
+        graph.nodes.each_value do |node|
+          node_class = node.props.dig(:node_class, :node_class)
+          case node_class
+          when 'org.graalvm.compiler.nodes.IfNode'
+            notes.add 'branches'
+          when 'org.graalvm.compiler.nodes.LoopBeginNode'
+            notes.add 'loops'
+          when 'org.graalvm.compiler.nodes.InvokeNode', 'org.graalvm.compiler.nodes.InvokeWithExceptionNode'
+            notes.add 'calls'
+          end
+        end
+
+        notes.add 'deopts' if graph.nodes[0].outputs.map(&:to)
+                                   .all? { |t| t.props.dig(:node_class, :node_class) == 'org.graalvm.compiler.nodes.DeoptimizeNode' }
```

Comment:
`deopts` is not very clear, the actual meaning is all branches deopt/all executions end up in a deopt, right?
Maybe `all deopt`?